pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1259,7 +1271,12 @@ func (configgen *ConfigGeneratorImpl) onVirtualOutboundListener(env *model.Envir
 		return ipTablesListener
 	}
 
-	filter := ipTablesListener.FilterChains[0].Filters[0]
+	// contains all filter chains except for the final passthrough/blackhole
+	initialFilterChain := ipTablesListener.FilterChains[:len(ipTablesListener.FilterChains)-1]
+
```

Comment:
Except? Shouldn’t it be Len-2?

Reply:
I think [:len] will select all of them so just need -1. seems to work, let me double check though